spec/acceptance/class_spec.rb

```diff
@@ -1,16 +1,63 @@
 require 'spec_helper_acceptance'
 
-describe 'nodejs class:', unless: UNSUPPORTED_PLATFORMS.include?(fact('osfamily')) do
-  it 'runs successfully' do
-    pp = <<-EOS
-    class { 'nodejs': }
-    if $::osfamily == 'RedHat' and $::operatingsystemrelease =~ /^5\.(\d+)/ {
-      class { 'epel': }
-      Class['epel'] -> Class['nodejs']
-    }
-    EOS
+describe 'nodejs class:', unless: UNSUPPORTED_PLATFORMS.include?(fact('os.family')) do
+  case fact('os.family')
+  when 'RedHat'
+    pkg_cmd = 'yum info nodejs | grep "^From repo"'
+  when 'Debian'
+    pkg_cmd = 'dpkg -s nodejs | grep ^Maintainer'
+  end
+
+  context 'default parameters' do
+    let(:pp) do
+      <<-EOS
+      class { 'nodejs': }
+      if $::osfamily == 'RedHat' and $::operatingsystemrelease =~ /^5\.(\d+)/ {
+        class { 'epel': }
+        Class['epel'] -> Class['nodejs']
+      }
+      EOS
+    end
+
+    it 'runs successfully' do
+      apply_manifest(pp, catch_failures: true)
+      apply_manifest(pp, catch_changes: true)
+    end
+
+    if %w[RedHat Debian].include? fact('os.family')
+      describe package('nodejs') do
+        it { is_expected.to be_installed }
+        it 'comes from the expected source' do
+          pkg_output = shell(pkg_cmd)
+          expect(pkg_output.stdout).to match 'nodesource'
+        end
+      end
+    end
+  end
+
+  context 'repo_class => epel', if: fact('os.family') == 'RedHat' do
+    let(:pp) do
+      <<-EOS
+        class { 'nodejs':
+          repo_class => '::epel',
+        }
+        class { 'epel': }
```

Comment:
I think these 2 lines are redundant (or should be): https://github.com/voxpupuli/puppet-nodejs/blob/8fc47fb1644ca1818da3cc834819286a93b7c20f/manifests/init.pp#L36-L39

Reply:
I wasn't sure about the history of that, so was hesitant to yank it otu; I will see if the test still works Ok with that removed, though, and commit if so.

We should also switch to 'contain' vs. anchor pattern (can do this in a separate PR)

Reply:
> We should also switch to 'contain' vs. anchor pattern (can do this in a separate PR)

+1